### PR TITLE
DOC: only single-polynomial fitting in np.polynomial.Polynomial.fit()

### PR DESCRIPTION
**Pull request by wrzadkow**
A fix of the documentation in `np.polynomial.Polynomial.fit()`, Issue #16380.

The discussion in this issue reached a conclusion not to implement the solution proposed in PR #16403, hence the documentation fix as a short-term solution instead.

Also mentioned in Issue #4554

### CODE REVIEW

File: numpy/polynomial/_polybase.py [hunk 1]
```diff
@@ -919,10 +919,8 @@ def fit(cls, x, y, deg, domain=None, rcond=None, full=False, w=None,
         ----------
         x : array_like, shape (M,)
             x-coordinates of the M sample points ``(x[i], y[i])``.
-        y : array_like, shape (M,) or (M, K)
-            y-coordinates of the sample points. Several data sets of sample
-            points sharing the same x-coordinates can be fitted at once by
-            passing in a 2D-array that contains one dataset per column.
+        y : array_like, shape (M,)
+            y-coordinates of the sample points.
```
Comment (rossbar) [inline numpy/polynomial/_polybase.py:923]:
```suggestion
            y-coordinates of the sample points ``(x[i], y[i])``.
```
Might as well add this extra wording to show consistency between the parameters, now that all of the extra wording is removed.

**Review by rossbar — COMMENTED** [review]
Thanks @wrzadkow , I think this is the right way to proceed given the discussion in #16380 . Just one little comment about the parameter string - feel free to take-it-or-leave-it!

Also, it might be worthwhile to add a Notes section that points to the `*fit` functions and highlights that they produce coefficient *arrays* and support fitting multiple data sets for `(M, K)`-shaped inputs.

**Comment by wrzadkow** [comment]
Thanks, @rossbar!

> it might be worthwhile to add a Notes section that points to the *fit functions and highlights that they produce coefficient arrays and support fitting multiple data sets for (M, K)-shaped inputs.

Do you mean here adding a Notes section about the `np.polynomial.polynomial.polyfit()` function? The function `np.polynomial.Polynomial._fit()` also works, but I guess it should be kept "private", i.e. not explicitly mentioned in the documentation, correct?

**Comment by rossbar** [comment]
> Do you mean here adding a Notes section about the np.polynomial.polynomial.polyfit() function? The function np.polynomial.Polynomial._fit() also works, but I guess it should be kept "private", i.e. not explicitly mentioned in the documentation, correct?

Yes, I was thinking of the individual fitting functions for each type of polynomial (e.g. `polyfit`, `chebfit`, `hermfit`, etc.). The naming convention is a little unfortunate in that it would be difficult to cover them clearly and succinctly.

File: numpy/polynomial/_polybase.py [hunk 2]
```diff
@@ -919,10 +919,8 @@ def fit(cls, x, y, deg, domain=None, rcond=None, full=False, w=None,
         ----------
         x : array_like, shape (M,)
             x-coordinates of the M sample points ``(x[i], y[i])``.
-        y : array_like, shape (M,) or (M, K)
-            y-coordinates of the sample points. Several data sets of sample
-            points sharing the same x-coordinates can be fitted at once by
-            passing in a 2D-array that contains one dataset per column.
+        y : array_like, shape (M,)
+            y-coordinates of the sample points ``(x[i], y[i])``.
```
Comment (rossbar) [inline numpy/polynomial/_polybase.py:923]:
```suggestion
            y-coordinates of the M sample points ``(x[i], y[i])``.
```
Oops, I missed that last time - for consistency's sake

**Review by rossbar — APPROVED** [review]
LGTM - thanks @wrzadkow ! You're welcome to add a Notes or See Also section to call out the module fitting functions, but I think this is fine as-is.

**Comment by wrzadkow** [comment]
Thanks again, @rossbar. After thinking it over, I have the impression that it would be hard for me to include these functions clearly and succintly, as you mention. Also after this documentation update it looks to me like in principle nobody should expect the function `Polynomial.fit()` to work for multi-polynomial case, which could reduce the demand for such case to be covered in the documentation.

**Comment by rossbar** [comment]
@charris does this documentation change seem reasonable to you given the ongoing discussion in #16380? It sounds like there may be some interest in potentially expanding the polynomial classes in the future, but in the meantime just modify the `fit` docstring to reflect the current (1D, single data-set) behavior.

**Pull request merged**